src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt

```diff
@@ -153,69 +175,68 @@ class NotificationService(
     * 10 means we can cover 99.9% of scenarios in one email.
     */
     private fun sendNotification(quantumId: String, notificationGroup: List<ShiftNotification>) {
-        val userPreference = userPreferenceService.getOrCreateUserPreference(quantumId)
+        log.info("Found ${notificationGroup.size} results for $quantumId")
 
-        data class Key(val shiftDate: LocalDate, val shiftType: String, val quantumId: String)
-
-        fun ShiftNotification.toKeyDuplicates() = Key(this.shiftDate, this.shiftType, this.quantumId.toUpperCase())
-
-        // Only send the latest notification for a shift if there are multiple
-        val mostRecentNotifications = notificationGroup
-                .groupBy { it.toKeyDuplicates() }
-                .map { (key, value) -> value.maxBy { it.shiftModified } }
-                .filterNotNull()
+        val userPreference = userPreferenceService.getOrCreateUserPreference(quantumId)
 
-        if (shouldSend(userPreference)) {
-            log.debug("Sending (${mostRecentNotifications.size}) notifications to ${userPreference.quantumId}, preference set to ${userPreference.commPref}")
-            mostRecentNotifications.sortedWith(compareBy { it.shiftDate }).chunked(10).forEach { chunk ->
-                when (val communicationPreference = CommunicationPreference.from(userPreference.commPref)) {
-                    CommunicationPreference.EMAIL -> {
-                        notifyClient.sendEmail(NotificationType.EMAIL_SUMMARY.value, userPreference.email, generateTemplateValues(chunk, communicationPreference), null)
+        if (!userHasSnoozedNotifications(userPreference.snoozeUntil)) {
+            // Only send the latest notification for a shift if there are multiple
+            notificationGroup
+                    .asSequence()
```

Comment:
https://blog.kotlin-academy.com/effective-kotlin-use-sequence-for-bigger-collections-with-more-than-one-processing-step-649a15bb4bf